### ISSUE TITLE
OpenMS via pyOpenMs on arm64?

### ISSUE THREAD

**Issue by fkromer** [OpenMS/OpenMS#4816]
I've tried to install OpenMS via pyOpenMS into a Docker container via "cross building" using `dockerx` with `docker buildx build --platform linux/arm64 ...`. Obviously the container shall be build for/run on an embedded board. Unfortunately I get an error message which indicates that pyopenms does not contain/provide OpenMS cross-compiled for/compiled on ARM architecture:

 `#9 392.6 ERROR: Could not find a version that satisfies the requirement pyopenms==2.4.0 (from -r requirements.txt (line 307)) (from versions: 2.3.0.linux-x86_64, 2.3.0.1.linux-x86_64, 2.3.0.2.linux-x86_64, 2.3.0.3.linux-x86_64)`.

Would it be possible to add multi architecture support to the Python package? Or a package exclusively for the arm64 architecture?

**Comment by jpfeuffer** [comment]
Hi,

To cross-compile pyOpenMS for arm, I would use the dockcross/manylinux2014-aarch64 containers.
Unfortunately, you have to expect some struggle since you will need to cross-compile all dependencies first (manually and/or update our contrib build scripts). Maybe you find some of the dependencies as ARM packages on apt though.

I tried it on a new branch with GitHub actions (https://github.com/OpenMS/OpenMS/runs/879689427?check_suite_focus=true) but Boost fails since it self-compiles its build system b2 with the cross-compiler already and then can't execute it (i.e. this needs some patch in OpenMS/contrib).
Also, Qt5 has to be compiled from scratch (which is still missing in the build scripts I used on GH Actions, since we usually download binaries).

So, it is not impossible but some work and unfortunately not a very high priority ATM. I could probably assist if you give it a try, otherwise I will probably do it bit by bit.

**Comment by fkromer** [comment]
> Hi,
>
> To cross-compile pyOpenMS for arm, I would use the dockcross/manylinux2014-aarch64 containers. Unfortunately, you have to expect some struggle since you will need to cross-compile all dependencies first (manually and/or > update our contrib build scripts). Maybe you find some of the dependencies as ARM packages on apt though.

Right, if one has to cross-compile it's always good to not being too optimistic :slightly_smiling_face: 

> I tried it on a new branch with GitHub actions (https://github.com/OpenMS/OpenMS/runs/879689427?check_suite_focus=true) but Boost fails since it self-compiles its build system b2 with the cross-compiler already and then can't execute it (i.e. this needs some patch in OpenMS/contrib).
Also, Qt5 has to be compiled from scratch (which is still missing in the build scripts I used on GH Actions, since we usually download binaries).
> 
> So, it is not impossible but some work and unfortunately not a very high priority ATM. I could probably assist if you give it a try, otherwise I will probably do it bit by bit.

Thanks a lot for your effort. The implicit Qt5 dependency surprised me a bit...  I decided to get rid of pyOpenMS in my ARM architecture runtime environment. This means this issue isn't of interest for me ATM. Obviously this means I cannot justify putting effort into this as well.

**Comment by jpfeuffer** [comment]
Yes, Qt5 was introduced for our GUI tools but was then used also for basic File and Web handling, so even with GUI=Off it will be needed for the library which underlies pyOpenMS.

Sorry to hear about the exclusion but I can understand. You can track this issue, it will get more important when the Macs switch to ARM.

**Comment by fkromer** [comment]
@jpfeuffer Alright. Thanks.
